src/session/session_api.c

```diff
@@ -830,6 +831,18 @@ __session_open_cursor(WT_SESSION *wt_session, const char *uri, WT_CURSOR *to_dup
         WT_ERR_MSG(
           session, EINVAL, "should be passed either a URI or a cursor to duplicate, but not both");
 
+    /*
+     * Disallow external cursor creation on a leaders ingest table if not configured with readonly
+     * configuration.
+     */
+    if (S2C(session)->layered_table_manager.leader && uri != NULL &&
+      WT_SUFFIX_MATCH(uri, ".wt_ingest") && !F_ISSET(session, WT_SESSION_INTERNAL)) {
```

Comment:
One question, do internal sessions take this path? If no, then `WT_SESSION_INTERNAL` should never be set.